lib/Chat/Parser/UserMention.php

```diff
@@ -108,12 +108,17 @@ public function parseMessage(Message $chatMessage): void {
 			// index of the mentions of that type.
 			$mentionParameterId = 'mention-' . $mention['type'] . $mentionTypeCount[$mention['type']];
 
+			$pattern = '/(^|\s)(' . '@' . $mention['id'] . ')(\b)/';
 			if (strpos($mention['id'], ' ') !== false || strpos($mention['id'], 'guest/') === 0) {
-				$placeholder = '@"' . $mention['id'] . '"';
-			} else {
-				$placeholder = '@' .  $mention['id'];
+				$pattern = '/(^|\s)(' . '@"' . preg_quote($mention['id'], '/') . '"' . ')(\b)?/';
 			}
-			$message = str_replace($placeholder, '{' . $mentionParameterId . '}', $message);
+
+			$message = preg_replace(
+				$pattern,
+				//'${1}' . $this->regexSafeUser($mention['id'], $displayName) . '${3}',
```

Comment:
Copied from https://github.com/nextcloud/server/blob/24d436cb600a725ba162a5387552a996a6fc486f/apps/comments/lib/Activity/Provider.php#L223
so would keep for consistancy to have matching search results in case someone ever does something with it